spec/compiler/normalize/array_literal_spec.cr

```diff
@@ -6,10 +6,20 @@ describe "Normalize: array literal" do
   end
 
   it "normalizes non-empty with of" do
-    assert_expand "[1, 2] of Int", "::Array(Int).build(2) do |__temp_1|\n  __temp_1[0] = 1\n  __temp_1[1] = 2\n  2\nend"
+    assert_expand "[1, 2] of Int8", <<-CR
+      __temp_1 = ::Array(Int8).unsafe_build(2)
+      __temp_1.to_unsafe[0] = 1
+      __temp_1.to_unsafe[1] = 2
+      __temp_1
```

Comment:
Could we change this to be:

```crystal
__temp_1 = ::Array(Int8).unsafe_build(2)
__temp_2 = __temp_1.to_unafe
__temp_2[0] = 1
__temp_2[1] = 2
__temp_1
```

That way the `to_unsafe` call would be needed to be computed just once.

Reply:
I would have expected that call to be inlined. But sure, I'll change it.

Reply:
Oh, actually it is inlined by the codegen because it's just returning an instance var. But the semantic phase will have to look it up once per each element, so maybe it's too much.